Library/ViewModels/ConfirmDetailsViewModel.swift

```diff
@@ -224,13 +230,81 @@ public class ConfirmDetailsViewModel: ConfirmDetailsViewModelType, ConfirmDetail
       context.map { $0.confirmationLabelHidden }
     )
 
-    self.configurePledgeSummaryViewControllerWithData = Signal.combineLatest(
+    let pledgeTotalSummaryData = Signal.combineLatest(
       projectAndConfirmationLabelHidden,
       pledgeTotal
     )
     .map(unpack)
     .map { project, confirmationLabelHidden, total in (project, total, confirmationLabelHidden) }
     .map(pledgeSummaryViewData)
+
+    self.configurePledgeSummaryViewControllerWithData = pledgeTotalSummaryData
+
+    // MARK: Pledge + Rewards Summary Table
+
+    self.pledgeRewardsSummaryViewHidden = Signal.zip(context, baseReward)
+      .map { context, reward in
+        if context.isAny(of: .pledge, .updateReward) {
+          return reward.isNoReward
+        }
+
+        return context.expandableRewardViewHidden
+      }
+
+    let shippingSummaryData = Signal.combineLatest(
+      selectedShippingRule.skipNil().map(\.location.localizedName),
+      project.map(\.stats.omitUSCurrencyCode),
+      project.map { project in
+        projectCountry(forCurrency: project.stats.currency) ?? project.country
+      },
+      allRewardsShippingTotal
+    )
+    .map(PledgeShippingSummaryViewData.init)
+
+    let bonusOrPledgeUpdatedAmount = self.pledgeAmountDataSignal.map { $0.amount }
+
+    self.configurePledgeRewardsSummaryViewWithData = Signal.combineLatest(
+      baseReward.map(\.isNoReward).filter(isFalse),
+      project,
+      rewards,
+      selectedQuantities,
+      shippingSummaryData,
+      bonusOrPledgeUpdatedAmount,
+      pledgeTotalSummaryData
+    )
+    .map { _, project, rewards, selectedQuantities, shippingSummaryData, bonusOrPledgeUpdatedAmount, pledgeTotalSummaryData -> (
+      PostCampaignRewardsSummaryViewData,
+      Double?,
+      PledgeSummaryViewData
+    ) in
+    guard let projectCurrencyCountry = projectCountry(forCurrency: project.stats.currency) else {
+      return (
+        PostCampaignRewardsSummaryViewData
+          .init(
```

Comment:
Nit: seems unusual to call `init` instead of just calling `PostCampaignRewardsSummaryViewData(...)`